packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -171,6 +174,40 @@ export function getSchemaForType(
   }
 
   if (type.kind === "Model") {
+    const httpPart = getHttpPart(program, type);
```

Comment:
Should we double check the bodyKind also?  https://github.com/witemple-msft/typespec/blob/witemple-msft/http-file-body/packages/http/src/types.ts#L486-L492

See [issue ](https://github.com/microsoft/typespec/issues/6530)here for httppart used in common content type.

Reply:
I had a look at this and it's a bit complicated to check the bodyKind since we don't have access to the operation definition here. I'd like to get this basic implementation checked in first to unblock the main use case and will open an issue to revisit this. Hopefully people won't be using HttpPart outside of MFD very often.

Reply:
https://github.com/Azure/autorest.typescript/issues/3144